gatsby/content/blog/2020/04/2020-04-06-selfhosting-matrix.mdx

```diff
@@ -0,0 +1,162 @@
+ * Enable TLS via LetsEncrypt on nginx, by: `apt install -y python3-certbot-nginx && certbot --nginx -d dangerousdemos.net -d riot.dangerousdemos.net -d matrix.dangerousdemos.net` (or whatever your domains are).
+ * You should be able to go to https://dangerousdemos.net at this point and see a page with valid HTTPS.
+
+### Synapse
+
+ * Then, install Synapse via Debian packages using the instructions at https://github.com/matrix-org/synapse/blob/master/INSTALL.md#debianubuntu (see below)
+   * You should specify the server name to be the domain you want in your matrix IDs - i.e. `dangerousdemos.net` in this example.
+   * Please report anonymous aggregate stats to us so we can gauge uptake and help justify funding for Matrix!
+
+```
+sudo apt install -y lsb-release wget apt-transport-https
+sudo wget -O /usr/share/keyrings/matrix-org-archive-keyring.gpg https://packages.matrix.org/debian/matrix-org-archive-keyring.gpg
+echo "deb [signed-by=/usr/share/keyrings/matrix-org-archive-keyring.gpg] https://packages.matrix.org/debian/ $(lsb_release -cs) main" |
+    sudo tee /etc/apt/sources.list.d/matrix-org.list
+sudo apt update
+sudo apt install matrix-synapse-py3
+```
+
+ * You should now be able to go to https://matrix.dangerousdemos.net and see a valid "It works! Synapse is running" page.
+ * Then, you should enable registration on your synapse by switching `enable_registration: true` in `/etc/matrix-synapse/homeserver.yaml` and restarting synapse via `systemctl restart matrix-synapse`.
+
+ * Now you need to tell the rest of Matrix how to find your server.  The easiest way to do this is to publish a file at https://dangerousdemos.net/.well-known/matrix/server which tells everyone the hostname and port where they can find the synapse for dangerousdemos.net - in this instance, it's `matrix.dangerousdemos.net:443`:
+
+```
+mkdir -p /var/www/dangerousdemos.net/.well-known/matrix
+cd /var/www/dangerousdemos.net/.well-known/matrix
+cat '{ "m.server": "matrix.dangerousdemos.net:443" }' > server
+```
+
+### Riot/Web
+
+ * Then, install Riot/Web.  Grab the latest .tgz release from https://github.com/vector-im/riot-web/releases.  You should check its GnuPG signature too:
+
+```
+mkdir /var/www/riot.dangerousdemos.net
+cd /var/www/riot.dangerousdemos.net
+wget https://github.com/vector-im/riot-web/releases/download/v1.5.15/riot-v1.5.15.tar.gz
+
+# check its GnuPG signature (particularly advisable, given Riot is what stores
+# your end-to-end encryption keys)
+apt install -y gpg
+wget https://github.com/vector-im/riot-web/releases/download/v1.5.15/riot-v1.5.15.tar.gz.asc
+
+# grab the signing key for the riot releases repository, ideally from a keyserver
+# e.g. https://keyserver.ubuntu.com/ and search for releases@riot.im.
+# This should give a keyID like 74692659bda3d940, which you can then import:
+gpg --keyserver keyserver.ubuntu.com --recv-keys 74692659bda3d940
```

Comment:
It might be smoother to just run `gpg --keyserver keyserver.ubuntu.com --search-keys releases@riot.im` instead, which does both the search and retrieval in a single step.